src/openrct2-ui/windows/Player.cpp

```diff
@@ -400,6 +400,13 @@ void window_player_overview_paint(rct_window* w, rct_drawpixelinfo* dpi)
 
 void window_player_overview_invalidate(rct_window* w)
 {
+    int32_t playerIndex = network_get_player_index((uint8_t)w->number);
+    if (playerIndex == -1)
+    {
+        window_close(w);
```

Comment:
I am not sure if this is valid, do any other windows close the window within `invalidate`? Perhaps the update event would be a better place.

Reply:
Theres another invalidate function where this is the happening as well but yeah this feels somewhat dirty. I'm not even entirely sure if this is really required more a fail safe to not access out of bounds data.

Reply:
Well I mean just keep the check and return but actually close the window in a safer event handler. But if another window already does this then I guess it isn't a problem. I just didn't know whether the same window would try to then be painted after being closed.

Reply:
Turns out window_player_statistics_update is already the update event, easy to overlook given that its a list of pointers. I'll just return like you suggested.